### PR TITLE
refact(es/minifier): merge `cond_init` with `reassigned`

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

And optimize following sutiation
```js
export function genElement(el, state) {
    if ('slot' === el.tag) return el1 = el, genChildren(el1);
    if (el.component) {
        var el1
        return 999;
    }
}
```
which rarely happens in hand written JS, but is often generated by swc merge variable pass

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs [hunk 1]
```diff
@@ -771,9 +773,11 @@ where
 
                 match &n.body {
                     Some(body) => {
+                        println!("function");
```
Comment (kdy1) [inline crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs:776]: Please remove this

File: crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs [hunk 2]
```diff
@@ -771,9 +773,11 @@ where
...
                         // We use visit_children_with instead of visit_with to bypass block scope
                         // handler.
                         body.visit_children_with(child);
+                        println!("functionend");
```
Comment (kdy1) [inline crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs:780]: And this

File: crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs [hunk 3]
```diff
@@ -297,9 +297,11 @@ where
 
     #[cfg_attr(feature = "debug", tracing::instrument(skip(self, n)))]
     fn visit_block_stmt(&mut self, n: &BlockStmt) {
+        println!("block");
         self.with_child(n.span.ctxt, ScopeKind::Block, |child| {
             n.visit_children_with(child);
-        })
+        });
+        println!("blockend");
```
Comment (kdy1) [inline crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs:304]: Also this

File: crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs [hunk 4]
```diff
@@ -297,9 +297,11 @@ where
 
     #[cfg_attr(feature = "debug", tracing::instrument(skip(self, n)))]
     fn visit_block_stmt(&mut self, n: &BlockStmt) {
+        println!("block");
```
Comment (kdy1) [inline crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs:300]: Also this

File: crates/swc_ecma_minifier/src/program_data.rs [hunk 5]
```diff
@@ -379,6 +374,8 @@ impl Storage for ProgramData {
         v.declared = true;
         // not a VarDecl, thus always inited
         if has_init || kind.is_none() {
+            println!("init {}", i.sym);
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/program_data.rs:377]: Also this

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_usage_analyzer --breaking

**Review by jridgewell — APPROVED** [review]

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**